BUG: test_doccer:test_decorator fails with Python 3.13 due to dedented docstring constants

### Describe your issue.

When I try to build scipy 1.11.3 in Fedora with Python 3.13.0a1+ I see the following test failure:

```
________________________________ test_decorator ________________________________
[gw1] linux -- Python 3.13.0 /usr/bin/python3

    @pytest.mark.skipif(DOCSTRINGS_STRIPPED, reason="docstrings stripped")
    def test_decorator():
        with suppress_warnings() as sup:
            sup.filter(category=DeprecationWarning)
            # with unindentation of parameters
            decorator = doccer.filldoc(doc_dict, True)
    
            @decorator
            def func():
                """ Docstring
                %(strtest3)s
                """
>           assert_equal(func.__doc__, """ Docstring
                Another test
                   with some indent
                """)
E           AssertionError: 
E           Items are not equal:
E            ACTUAL: 'Docstring\nAnother test\n   with some indent\n'
E            DESIRED: ' Docstring\n            Another test\n               with some indent\n            '

scipy/misc/tests/test_doccer.py:86: AssertionError
```

This is likely caused by https://github.com/python/cpython/issues/81283 -- *Automatically dedent docstring constants by default*.

I tried to get around this by asserting the docstrings are equal to manual dostrings, e.g.:

```python
        @decorator
        def func():
            """ Docstring
            %(strtest3)s
            """

        def expected():
            """ Docstring
            Another test
               with some indent
            """
        assert_equal(func.__doc__, expected.__doc__)
```

But I think the dosctring of the original function is dedented first and the substitution happens after; hence the results are not the same as dedented manually constructed docstrings.

I could add if sys.version around the expected value, but I am unsure if that is the proper fix. Before I submit a PR that makes changes to the test, I need to understand how we want this to behave on Python 3.13.

### Reproducing Code Example

```python
run the tests
```


### Error message

```shell
see above
```


### SciPy/NumPy/Python version and system information

```shell
I am no longer able to paste the command above. This is on scipy 1.11.3 with numpy 1.26.0 and Python built from the main branch.
```

--- a/scipy/misc/tests/test_doccer.py
+++ b/scipy/misc/tests/test_doccer.py
@@ -72,7 +72,6 @@ def test_docformat():
 
 
 @pytest.mark.skipif(DOCSTRINGS_STRIPPED, reason="docstrings stripped")
-@pytest.mark.skipif(sys.version_info >= (3, 13), reason='it fails on Py3.13')
 def test_decorator():
     with suppress_warnings() as sup:
         sup.filter(category=DeprecationWarning)
@@ -84,23 +83,31 @@ def test_decorator():
             """ Docstring
             %(strtest3)s
             """
-        assert_equal(func.__doc__, """ Docstring
+
+        def expected():
+            """ Docstring
             Another test
                with some indent
-            """)
+            """
+        assert_equal(func.__doc__, expected.__doc__)
 
         # without unindentation of parameters
-        decorator = doccer.filldoc(doc_dict, False)
+
+        # The docstring should be unindented for Python 3.13+
+        # because of https://github.com/python/cpython/issues/81283
+        decorator = doccer.filldoc(doc_dict, False if sys.version_info < (3, 13) else True)
 
         @decorator
         def func():
             """ Docstring
             %(strtest3)s
             """
-        assert_equal(func.__doc__, """ Docstring
+        def expected():
+            """ Docstring
                 Another test
                    with some indent
-            """)
+            """
+        assert_equal(func.__doc__, expected.__doc__)
 
 
 @pytest.mark.skipif(DOCSTRINGS_STRIPPED, reason="docstrings stripped")

--- a/scipy/misc/tests/test_doccer.py
+++ b/scipy/misc/tests/test_doccer.py
@@ -95,7 +95,8 @@ def test_decorator():
 
         # The docstring should be unindented for Python 3.13+
         # because of https://github.com/python/cpython/issues/81283
-        decorator = doccer.filldoc(doc_dict, False if sys.version_info < (3, 13) else True)
+        decorator = doccer.filldoc(doc_dict, False if \
+                                   sys.version_info < (3, 13) else True)
 
         @decorator
         def func():

